core/standards.md

```diff
@@ -0,0 +1,80 @@
+
+This repo attempts to document our standards, other sources include (non-exhaustive list):
+
+- [MIPS best practices](https://github.com/makerdao/mips-best-practices)
+
+## Coding standards we expect to see in core contracts
+
+### Permissions
+
+Permissioning in the contract should be handled by `wards` and the `wards` mapping should occupy storage slot `0x0`
+
+### Contract ordering
+
+Following and expanding on [Solidity Order of layout recommendations](https://docs.soliditylang.org/en/stable/style-guide.html#order-of-layout) our contracts are generally laid out in this way:
+
+1. State variables (including wards and math)
+
+    a. constants then immutables then storage variables
+
+    b. wards should be storage slot `0x0`
+2. Events
+3. Getter functions (i.e. to make reading structs easier)
+
+    a. these may be placed at the end of the file if they are extensive to improve readability
+4. Constructor
+5. Modifiers
+6. Internal Math variables and functions
+7. Wards functions (i.e. `rely`/`deny`)
+8. Fileable functions
+9. Other Internal functions
+10. Other External functions
+
+### Spacing
+
+We align variables with extra whitespace. You will get PR comments if you are off by even one space.
+
+### Naming
+
+We prefer to try to have **Daiwanese** naming patterns (3 or 4 letter variables and functions).
+
+If we are avoiding variable shadowing, we add a `_` to the end of the variable names.
+
+If we are dealing with internal variables or functions we add the `_` to the front of the name.
+
+If we need to interact with external contracts, we prefer to have a minimal static interface using the naming convention of `<ContractName>Like`. (These should be declared at the top of the contract file.)
+
+### Dependencies
+
+We try to keep our repos relatively dependency free.  We do not rely on external contract libraries (i.e. Open Zeppelin) and try on only import submodules that are absolutely necessary as this keeps the cascading updates to a minimum.
+
+### Antipatterns
+
+In order to limit contract complexity, we also tend to stay away from inheritance patterns.
+
+We believe that upgradable proxies are an antipattern.  If you need upgradability, it should be handled with modular design and permission given to the Pause Proxy. There are rare exceptions to this such as where there is a user storage mapping and high chance of needing to upgrade (e.g. cropper or charter).
+
+### Documentation
+
+Repositories should have a README file that outline the purpose, functionality and how to use the contract.  In addition to this external/public functions should have [NatSpec comments](https://docs.soliditylang.org/en/latest/style-guide.html#natspec).
```

Comment:
### Deployment

Contracts being deployed should follow existing patterns in terms of optimisation settings and solc versions, adapted where appropriate. Deployments with rare exception should not change their optimisation settings between release versions. 
(Here I am talking about spells being an exception)

Reply:
So I feel like this wording is a little non-specific, how do you feel about what i included here: https://github.com/makerdao/pe-checklists/pull/2/commits/be221016ee1746864099252b6760156ff4e16211

Reply:
Good, "In most cases constants and immutables can be declared internal" should specify that all constants and vars *must* have a visibility modifier set and it can't be default/blank